src/trans/cpu/external/gath_grid.F90

```diff
@@ -103,25 +103,25 @@ SUBROUTINE GATH_GRID(PGPG,KPROMA,KFGATHG,KTO,KRESOL,PGP)
 ENDDO
 
 IUBOUND=UBOUND(PGP)
-IF(IUBOUND(1) < IPROMA) THEN
-  WRITE(NOUT,*)'GATH_GRID:FIRST DIM. OF PGP TOO SMALL ',IUBOUND(1),IPROMA
-  CALL ABORT_TRANS('GATH_GRID:FIRST DIMENSION OF PGP TOO SMALL ')
+IF(IUBOUND(1) /= IPROMA) THEN
+  WRITE(NOUT,*)'GATH_GRID:FIRST DIMENSION OF PGP /= IPROMA',IUBOUND(1),IPROMA
+  CALL ABORT_TRANS('GATH_GRID:FIRST DIMENSION OF PGP /= IPROMA')
 ENDIF
-IF(IUBOUND(2) < KFGATHG) THEN
-  WRITE(NOUT,*)'GATH_GRID:SEC. DIM. OF PGP TOO SMALL ',IUBOUND(2),KFGATHG
-  CALL ABORT_TRANS('GATH_GRID:SECOND DIMENSION OF PGP TOO SMALL ')
+IF(IUBOUND(2) /= KFGATHG) THEN
+  WRITE(NOUT,*)'GATH_GRID:SEC. DIMENSION OF PGP /= KFGATHG',IUBOUND(2),KFGATHG
+  CALL ABORT_TRANS('GATH_GRID:SECOND DIMENSION OF PGP /= KFGATHG')
 ENDIF
-IF(IUBOUND(3) < IGPBLKS) THEN
-  WRITE(NOUT,*)'GATH_GRID:THIRD DIM. OF PGP TOO SMALL ',IUBOUND(3),IGPBLKS
-  CALL ABORT_TRANS('GATH_GRID:THIRD DIMENSION OF PGP TOO SMALL ')
+IF(IUBOUND(3) /= IGPBLKS) THEN
+  WRITE(NOUT,*)'GATH_GRID:THIRD DIMENSION OF PGP /= IGPBLKS',IUBOUND(3),IGPBLKS
+  CALL ABORT_TRANS('GATH_GRID:THIRD DIMENSION OF PGP /= IGPBLKS')
 ENDIF
 
 IF(IFRECV > 0) THEN
   IF(.NOT.PRESENT(PGPG)) THEN
     CALL ABORT_TRANS('GATH_GRID:PGPG MISSING')
   ENDIF
-  IF(UBOUND(PGPG,1) < D%NGPTOTG) THEN
-    CALL ABORT_TRANS('GATH_GRID:FIRST DIMENSION OF PGPG TOO SMALL')
+  IF(UBOUND(PGPG,1) /= D%NGPTOTG) THEN
+    CALL ABORT_TRANS('GATH_GRID:FIRST DIMENSION OF PGPG /= D%NGPTOTG')
   ENDIF
  IF(UBOUND(PGPG,2) < IFRECV) THEN
```

Comment:
I guess this is `<` on purpose because we don't know how many fields are being gathered? If that's the case, I am fine fine with this (since this is the last dimension), but we might want to give a better error message for this scenario, i.e. tell the user please increase the size of PGPG to XXX (now YYYY), or at lest write this into NOUT.

Reply:
But here we're talking about `NGPTOTG` -> the global number of grid points, not the number of fields that we're gathering. I agree that the previous check was probably innocuous, but is there any case where someone would intentionally pass a `PGPG` array that has more elements than `NGPTOTG` in the first dimension?

Reply:
the comment is on line 126 :) IFRECV, essentially I was asking if line 126, you want to keep the < or if we want to change it to /=. With NGPTOTG I am absolutely with you, noone should ever pass an array with not NGPTOTG, that will just not be consistent, and the library user can handle this.

Reply:
Oh I see. I think it should be `/=`. As with the other cases, I can't think of a reason why you'd pass an array with an oversized field dimension.